### ISSUE TITLE
builtin in

### ISSUE THREAD

**Issue by floam** [fish-shell/fish-shell#9348]
I'd like a cousin of `contains` (or `contains_seq`?), that considers stdin. Usage would be comprehensible as `command needle | in $haystack …`

At first this was a desire to make `builtin contains` or `contains_seq` accept stdin for the needle, and take $argv as the haystack, but verb is all wrong.

```console
> uname | in Linux Darwin
> and echo "that's a platform"
```

**Comment by faho** [comment]
This is quite similar to a bunch of other tools, including `string match`, `contains`, `case`,...

Why should this exist instead of augmenting one of those?

How should it work with multiple elements?

Do you have a better usage example? Because your current one is just

```fish
contains (uname) Linux Darwin
and echo "that's a platform
```

At which point I'd lean towards not adding another builtin with a synonym-name that can do slightl different things.

**Comment by floam** [comment]
It'd work like `contains_seq`, but not be slow as hell.

**Comment by floam** [comment]
> echo b\nc | in a b c d && echo true
true

**Comment by floam** [comment]
I think the best way to describe this, is probably: do a good version of `contains_seq` without the `--` flaw, and call it `in`.

**Comment by faho** [comment]
> echo b\nc | in a b c d && echo true
true

Oh, so you read the needle via stdin and look for that in the arguments.

I'd naively have expected the other way around, but that's probably not too important.

> call it in

How about making it an option to `contains`, which already is a builtin, already does containment tests and already takes options?

```fish
echo b\nc | contains --stdin a b c d
```

That also has the advantage that you don't need to explain the difference between `contains` and `in` to people, and you don't have to explain that `in` has two meanings (one with `for`).

**Comment by floam** [comment]
The way I mocked it up was just:

```fish
function in -d "true if stdin in argv"
                                            read -la list
                                            contains_seq $list -- $argv
                                        end
```

but that'll act weird if e.g. someone used it where `--` could occur. I noticed it is slow.

```console
time echo a\nb | in a b c d e f g

________________________________________________________
Executed in    2.10 millis    fish           external
   usr time  580.00 micros  580.00 micros    0.00 micros
   sys time  172.00 micros  172.00 micros    0.00 micros

floam@MBP ~/f/fish-shell (xcrun)> time contains a a b c d e f g

________________________________________________________
Executed in    3.00 micros    fish           external
   usr time    7.00 micros    7.00 micros    0.00 micros
   sys time   10.00 micros   10.00 micros    0.00 micros
```

**Comment by floam** [comment]
> Oh, so you read the needle via stdin and look for that in the arguments.

Yeah, cognitively it's using the pipe to enable constructs "if x in y", aware we can't have a keyword `in` the way Python does.

**Comment by faho** [comment]
Okay, this didn't happen, I'm gonna say it's not planned.
